src/test/Assets/TestProjects/TestWindowsOsShimsApp/Program.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace TestWindowsOsShimsApp
+{
+    public static class Program
+    {
+        public static void Main(string[] args)
+        {
+            Console.WriteLine("Hello World!");
+            Console.WriteLine(string.Join(Environment.NewLine, args));
+            Console.WriteLine($"Framework Version:{GetFrameworkVersionFromAppDomain()}");
+
+        #if WINDOWS
```

Comment:
Do you need the `#ifdef`?  It looks like you only ever run this on Windows.

Reply:
It is not needed, but I personally would prefer to keep it - makes it more explicit. I don't like code which doesn't run, but also doesn't work at all.